port/cpl_string.h

```diff
@@ -614,6 +646,191 @@ extern "C++"
      * with CPLFree */
     using CPLCharUniquePtr = std::unique_ptr<char, CPLFreeReleaser>;
 
+    namespace cpl
+    {
+
+    /*! @cond Doxygen_Suppress */
+    /** Iterator for a CSLConstList */
+    struct CPL_DLL CSLIterator
+    {
+        using iterator_category = std::input_iterator_tag;
+        using difference_type = std::ptrdiff_t;
+        using value_type = const char *;
+        using pointer = value_type *;
+        using reference = value_type &;
+
+        CSLConstList m_papszList = nullptr;
+        bool m_bAtEnd = false;
+
+        inline const char *operator*() const
+        {
+            return *m_papszList;
+        }
+        inline CSLIterator &operator++()
```

Comment:
Should we add a clang-tidy rule to add a line break between methods ?
maybe  (untested):
SeparateDefinitionBlocks: Always
EmptyLineBeforeAccessModifier: LogicalBlock

Reply:
> Should we add a clang-tidy rule to add a line break between methods ?

yes, sounds like a good idea